src/components/modal/report/form-step.jsx

```diff
@@ -0,0 +1,89 @@
+const bindAll = require('lodash.bindall');
+const PropTypes = require('prop-types');
+const React = require('react');
+const FormattedMessage = require('react-intl').FormattedMessage;
+const classNames = require('classnames');
+
+const Form = require('../../forms/form.jsx');
+const Button = require('../../forms/button.jsx');
+const Spinner = require('../../spinner/spinner.jsx');
+const FlexRow = require('../../flex-row/flex-row.jsx');
+require('../../forms/button.scss');
+
+/**
+ * Step to be used in a form progression. Provides wrapping form element,
+ * renders children input elements, then provides a next button row
+ * that responds to form validation and submission spinner.
+ */
+class FormStep extends React.Component {
+    constructor (props) {
+        super(props);
+        this.state = {
+            valid: false
+        };
+        bindAll(this, [
+            'handleValid',
+            'handleInvalid'
+        ]);
+    }
+    handleValid () {
+        this.setState({valid: true});
+    }
+    handleInvalid () {
+        this.setState({valid: false});
+    }
+    render () {
+        const {onNext, children, isWaiting, nextLabel} = this.props;
+        // Allow submit to be force-enabled by submitEnabled prop. Otherwise use form validation,
```

Comment:
I find this comment hard to follow.

How about something like:

// Submit button is enabled if form isn't already submitting, and either the form passes validation, or the submitEnabled prop is true.
// This lets submitEnabled prop override validation.

Reply:
Done